Libraries/SparkFun_ATECCX08a_Arduino_Library/src/SparkFun_ATECCX08a_Arduino_Library.h

```diff
@@ -0,0 +1,170 @@
+
+  Written by Pete Lewis @ SparkFun Electronics, August 5th, 2019
+
+  The IC uses I2C and 1-wire to communicat. This library only supports I2C.
+
+  https://github.com/sparkfun/SparkFun_ATECCX08A_Arduino_Library
+
+  Do you like this library? Help support SparkFun. Buy a board!
+
+  Development environment specifics:
+  Arduino IDE 1.8.1
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#if (ARDUINO >= 100)
+#include "Arduino.h"
+#else
+#include "WProgram.h"
+#endif
+
+#include "i2c_t3.h"
+
+#define ATECC608A_ADDRESS_DEFAULT 0x60 //7-bit unshifted default I2C Address
+// 0x60 on a fresh chip. note, this is software definable
+
+// WORD ADDRESS VALUES
+// These are sent in any write sequence to the IC.
+// They tell the IC what we are going to do: Reset, Sleep, Idle, Command.
+#define WORD_ADDRESS_VALUE_COMMAND 	0x03	// This is the "command" word address, 
+//this tells the IC we are going to send a command, and is used for most communications to the IC
+#define WORD_ADDRESS_VALUE_IDLE 0x02 // used to enter idle mode
+
+// COMMANDS (aka "opcodes" in the datasheet)
+#define COMMAND_OPCODE_INFO 	0x30 // Return device state information.
+#define COMMAND_OPCODE_LOCK 	0x17 // Lock configuration and/or Data and OTP zones
+#define COMMAND_OPCODE_RANDOM 	0x1B // Create and return a random number (32 bytes of data)
+#define COMMAND_OPCODE_READ 	0x02 // Return data at a specific zone and address.
+#define COMMAND_OPCODE_WRITE 	0x12 // Return data at a specific zone and address.
+#define COMMAND_OPCODE_SHA 		0x47 // Computes a SHA-256 or HMAC/SHA digest for general purpose use by the system.
+#define COMMAND_OPCODE_GENKEY 	0x40 // Creates a key (public and/or private) and stores it in a memory key slot
+#define COMMAND_OPCODE_NONCE 	0x16 // 
+#define COMMAND_OPCODE_SIGN 	0x41 // Create an ECC signature with contents of TempKey and designated key slot
+#define COMMAND_OPCODE_VERIFY 	0x45 // takes an ECDSA <R,S> signature and verifies that it is correctly generated from a given message and public key
+#define COMMAND_OPCODE_ECDH		0x43 // ECDH pre-master
+#define COMMAND_OPCODE_AES_ECB	0x51 // AES_ECB 
+
+
+// Lock command PARAM1 zone options (aka Mode). more info at table on datasheet page 75
+// 		? _ _ _  _ _ _ _ 	Bits 7 verify zone summary, 1 = ignore summary and write to zone!
+// 		_ ? _ _  _ _ _ _ 	Bits 6 Unused, must be zero
+// 		_ _ ? ?  ? ? _ _ 	Bits 5-2 Slot number (in this example, we use slot 0, so "0 0 0 0")
+// 		_ _ _ _  _ _ ? ? 	Bits 1-0 Zone or locktype. 00=Config, 01=Data/OTP, 10=Single Slot in Data, 11=illegal
+
+#define LOCK_MODE_ZONE_CONFIG 			0b10000000
```

Comment:
if there's a reason to have the values not in a column, ok, but it looks messy
lin 65 - 84